core.py

```diff
@@ -98,46 +129,70 @@ def enable_motors(self, enable):
         if self.arduino:
             self.arduino.enable_motors(enable)
 
-    def throttle(self, left_speed, right_speed):
+    def throttle(self,
+                 left_speed,
+                 right_speed,
+                 left_servo=ServoEnum.LEFT_MOTOR_ESC,
+                 right_servo=ServoEnum.RIGHT_MOTOR_ESC):
         """ Send motors speed value in range [-1,1]
             where 0 = neutral """
 
         # Calculate microseconds from command speed
-        left_micros = self.left_servo.micros(left_speed)
-        right_micros = self.right_servo.micros(right_speed)
+        left_micros = 0
+        right_micros = 0
+        try:
+            if left_servo != ServoEnum.SERVO_NONE:
+                left_micros = self.servos[left_servo][0].micros(left_speed)
+            if right_servo != ServoEnum.SERVO_NONE:
+                right_micros = self.servos[right_servo][0].micros(right_speed)
+        except:
```

Comment:
I guess that depends on what failure mode you're expecting. `self.servos[left_servo]` would fail with an `IndexError`, or if you *do* happen to get something back, but it doesn't have a `micros` property, then that would fail with an `AttributeError`.

What were you attempting to trap out here? Because at the moment, your calling code doesn't know that there was a failure, because all you're doing is `print()`ing and then continuing (i.e. returning `None`). Another thing to consider is either not handling the error here, and catching it in the calling code, or handling it, and raising a custom Exception which might be more meaningful in the calling code?